File: complete/src/main/java/com/example/restservice/RestServiceApplication.java
package com.example.restservice;

// SpringApplication: Spring Bootアプリケーションを起動するためのクラス
import org.springframework.boot.SpringApplication;
// SpringBootApplication: Spring Bootの自動設定を有効にするアノテーション
import org.springframework.boot.autoconfigure.SpringBootApplication;

/*
 * RestServiceApplication.java - アプリケーションのエントリーポイント（起動点）
 *
 * 【解説】
 * @SpringBootApplication は、以下の3つのアノテーションを1つにまとめたものです：
 *   1. @Configuration      : このクラスがBean（部品）の定義を含むことを示す
 *   2. @EnableAutoConfiguration : Spring Bootが依存ライブラリに基づいて
 *                                  自動的に設定を行うことを有効にする
 *   3. @ComponentScan      : このパッケージ以下のクラスを自動的に検出・登録する
 *                            （例: @RestController が付いたクラスを見つける）
 *
 * Javaプログラムは main() メソッドから実行が始まります。
 * SpringApplication.run() を呼ぶことで、組み込みWebサーバー（Tomcat）が起動し、
 * HTTPリクエストを受け付ける準備が整います。
 * デフォルトではポート8080で起動します（http://localhost:8080）。
 */
@SpringBootApplication
public class RestServiceApplication {

  // Javaプログラムのエントリーポイント（最初に実行されるメソッド）
  // args: コマンドライン引数（起動時にオプションを渡すために使用）
  public static void main(String[] args) {
    // Spring Bootアプリケーションを起動する
    // 第1引数: メインクラス（このクラス自身）
    // 第2引数: コマンドライン引数をそのまま渡す
    SpringApplication.run(RestServiceApplication.class, args);
  }

}

File: initial-kotlin/src/main/kotlin/com/example/restservice/RestServiceApplication.kt
package com.example.restservice

// SpringBootApplication: Spring Bootの自動設定を有効にするアノテーション
import org.springframework.boot.autoconfigure.SpringBootApplication
// runApplication: Kotlin用のSpring Boot起動ヘルパー関数
import org.springframework.boot.runApplication

/*
 * RestServiceApplication.kt - アプリケーションのエントリーポイント（起動点）（Kotlin版・初期状態）
 *
 * 【解説】
 * これはSpring Bootアプリケーションの「初期状態（initial）」のKotlin版です。
 * まだREST APIのエンドポイント（Greeting, GreetingController）は実装されていません。
 * complete-kotlin/ ディレクトリに完成版があります。
 *
 * @SpringBootApplication: Spring Bootの中核アノテーション。
 *   自動設定、コンポーネントスキャン、設定クラスの機能を兼ね備えています。
 *
 * Kotlinではクラス本体が空の場合、{} を省略できます。
 * main関数はクラスの外（トップレベル）に定義します。
 */
@SpringBootApplication
class RestServiceApplication

// Kotlinプログラムのエントリーポイント
// *args: スプレッド演算子で配列を個別の引数に展開
fun main(args: Array<String>) {
  runApplication<RestServiceApplication>(*args)
}
